File: workspace/cmms/cmms-web-app/src/app/app-auth/models/employeeProfile.ts
import {RoleName} from "./roleName";

export class EmployeeProfile {
    id?: number;
  username: string;
    name: string;
    surname: string;
    email: string;
    roleName: RoleName;
 //   password?: string;

  // private Long id;
  // private String username;
  // private String name;
  // private String surname;
  // private String email;
  // private RoleName roleName;

//
// {
//   "id": 20,
//   "username": "monika",
//   "name": "ownrflkjw",
//   "surname": "jmlkjmslk",
//   "email": "jhnwefvk@wp.pl",
//   "roleName": "EMPLOYEE"
// }


  constructor(id: number, username: string, name: string, surname: string, email: string, roleName: RoleName) {
    this.id = id;
    this.username = username;
    this.name = name;
    this.surname = surname;
    this.email = email;
    this.roleName = roleName;
  }
}

File: workspace/cmms/src/main/java/com/example/polls/payload/EmployeeProfile.java
package com.example.polls.payload;
import com.example.polls.model.RoleName;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class EmployeeProfile {

  private Long id;
  private String username;
  private String name;
  private String surname;
  private String email;
  private RoleName roleName;

  public EmployeeProfile() {
  }

  public EmployeeProfile(Long id, String username, String name, String surname, String email, RoleName roleName) {
    this.id = id;
    this.username = username;
    this.name = name;
    this.surname=surname;
    this.email=email;
    this.roleName=roleName;
  }


}
